src/tools/utils.ts

```diff
@@ -270,3 +270,8 @@ export const detectInsufficientScopeError = async <T>(
 export function sleep(ms: number): Promise<void> {
   return new Promise((resolve) => setTimeout(resolve, ms));
 }
+
+export const isDeprecatedError = (err: { message: string; statusCode: number }): boolean => {
+  if (!err) return false;
+  return !!(err.statusCode === 403 || err.message?.includes('deprecated'));
```

Comment:
I'm making an assumption here that the error message will be a 403 and include the word "deprecated". It _should_ work because early drafts show the message to be `'Insufficient privileges to use this deprecated feature'` but is worth noting because this error is not finalized yet.